### PR TITLE
[A3]Fix SnakePro vulnerability and add HTTPS support

### PR DESCRIPTION
**Pull request by danilogila**
Fix SnakePro vulnerability and add HTTPS support in order to protect user sensitive data.

### CODE REVIEW

File: owasp-top10-2017-apps/a3/snake-pro/app/api/routes.go [hunk 1]
```diff
@@ -55,6 +56,9 @@ func Register(c echo.Context) error {
 	userData.UserID = newGUID1.String()
 	userData.HighestScore = 0
 
+	password, err := bcrypt.GenerateFromPassword([]byte(userData.Password), 14)
```
Comment (Krlier) [inline owasp-top10-2017-apps/a3/snake-pro/app/api/routes.go]: @danilogila could you please check if `bcrypt.GenerateFromPassword([]byte(userData.Password), 14)` returns an error?

File: owasp-top10-2017-apps/a3/snake-pro/app/main.go [hunk 2]
```diff
@@ -71,10 +72,26 @@ func main() {
...
+			</code>
+		`
+		return c.HTML(http.StatusOK, fmt.Sprintf(format, req.Proto, req.Host, req.RemoteAddr, req.Method, req.URL.Path))
+	})
+
 	echoInstance.GET("/healthcheck", api.HealthCheck)
 	echoInstance.POST("/register", api.Register)
 	echoInstance.POST("/login", api.Login)
 	echoInstance.GET("/login", api.PageLogin)
+	echoInstance.Logger.Fatal(echoInstance.StartTLS(":10033", "cert.pem", "key.pem"))
```
Comment (Krlier) [inline owasp-top10-2017-apps/a3/snake-pro/app/main.go]: This line should be at after all routes declarations, and just be careful, you're starting the API once here and twice in `line 106`

Comment (danilogila) [reply owasp-top10-2017-apps/a3/snake-pro/app/main.go]: Thanks, Daniel. I fixed it at my last PR.

File: owasp-top10-2017-apps/a3/snake-pro/app/api/routes.go (hunk 1 above)
Comment (danilogila) [reply owasp-top10-2017-apps/a3/snake-pro/app/api/routes.go]: It didn't return an error. I added an return statement at my last PR.

**Comment by danilogila** [comment]
> You're almost there, @danilogila! I couldn't get the game to work, after registering and logging in I'm getting this error:
> 
> image
> 
> Could you please fix it?

@Krlier Fixed it. Could you take a look?

File: owasp-top10-2017-apps/a3/snake-pro/app/api/routes.go (hunk 1 above)
Comment (rafaveira3) [reply owasp-top10-2017-apps/a3/snake-pro/app/api/routes.go]: Although it did not return an error, for security reasons, it always good to add instructions if this error occurs (`!= nil`)

**Comment by rafaveira3** [comment]
Good job, @danilogila ! Your new code mitigates A3 vulnerabilities! 👏🏻🎉

**Pull request closed without merging**